newton/examples/elastic_rod/rod_mesher.py

```
@@ -0,0 +1,412 @@
+    radius: float,
+):
+    """Generate tube mesh for one rod. Shared by single and batched kernels."""
+    if num_points < 2:
+        return
+
+    # Initialise Bishop frame from first edge
+    w = wp.normalize(positions[pos_offset + 1] - positions[pos_offset])
+    fm = _basis_from_direction(w)
+    # Store frame columns at frame_offset (9 floats per rod)
+    fo = frame_offset
+    frames[fo + 0] = fm[0, 0]; frames[fo + 1] = fm[1, 0]; frames[fo + 2] = fm[2, 0]
```

Ruff (0.15.4)

[error] Multiple statements on one line (semicolon)

(E702)